### ISSUE TITLE
Resilience4j's TimeLimiter triggered after feign response

### ISSUE THREAD

**Issue by DevDengChao** [spring-cloud/spring-cloud-openfeign#582]
**Describe the bug**
Like the title said. I found that Resilience4j's TimeLimiter always triggered even feign client received response when feign method's return type is not `void`.

I'm using:
OpenJDK 1.8.0_292-b10 and 11.0.11.9-hotspot
Spring boot 2.5.2
Spring cloud 2020.0.3

**Sample**
Here: https://github.com/DevDengChao/resilience4j-time-limiter-triggered-after-feign-response

**Comment by DevDengChao** [comment]
Any progress =-=?

**Comment by DevDengChao** [comment]
Maybe I should submit this at https://github.com/spring-cloud/spring-cloud-circuitbreaker 😆

**Comment by cbezmen** [comment]
Hi @DevDengCha,
If you share example minimal repo which describes your bug, I will try to help you.

**Comment by DevDengChao** [comment]
> Hi @DevDengCha,
> If you share example minimal repo which describes your bug, I will try to help you.

Here: https://github.com/DevDengChao/resilience4j-time-limiter-triggered-after-feign-response , just download it and run the application and you'll reproduce this bug.

**Comment by idd-esousa** [comment]
Any updates? Has the bug been resolved?

**Comment by chenzhongya** [comment]
I have the same problem
 

 <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-starter-circuitbreaker-resilience4j</artifactId>
            <version>2.0.2</version>
  </dependency>


yaml:

feign:
  circuitbreaker:
    enabled: true
  httpclient:
    enabled: true
    disable-ssl-validation: true
  client:
    config:
      default:
        connectTimeout: 3100
        readTimeout: 3100
        loggerLevel: basic
resilience4j:
  thread-pool-bulkhead:
    instances:
       deviceRestartClinet:
       coreThreadPoolSize: 10
  timelimiter:
    instances:
      deviceRestartClinet:
        timeout-duration: 10s
        cancelRunningFuture: false

![image](https://user-images.githubusercontent.com/34332108/150762122-3f2961a1-8f9a-4934-8fda-efd6bdf167a7.png)
![image](https://user-images.githubusercontent.com/34332108/150762216-b3b3ca35-8358-4a35-af21-b0ba4de10b30.png)

**Comment by DevDengChao** [comment]
@chenzhongya  你可以用 ``` 包裹 yaml 内容, 渲染后的格式会更容易阅读一些.

eg, input:
<pre>
```yaml
feign:
  circuitbreaker:
    enabled: true
```
</pre>

output:
```yaml
feign:
  circuitbreaker:
    enabled: true
```

**Comment by DevDengChao** [comment]
New progress, see https://github.com/spring-cloud/spring-cloud-circuitbreaker/issues/120#issuecomment-1219968266

**Comment by OlgaMaciaszek** [comment]
Thanks @DevDengChao; I understand this workaround is working for you?

**Comment by DevDengChao** [comment]
@OlgaMaciaszek Yes, it works, and I updated the example repository, you can try it to confirm that.

**Comment by DevDengChao** [comment]
Should we close this issue now ?
